src/main/java/macrobase/analysis/contextualoutlier/ContextualOutlierDetector.java

```diff
@@ -97,128 +412,403 @@ public void searchContextualOutliers(List<Datum> data, double zScore){
 	private List<LatticeNode> levelUpLattice(List<LatticeNode> latticeNodes, List<Datum> data){
 		
 		//sort the subspaces by their dimensions
+		Stopwatch sw = Stopwatch.createUnstarted();
+		
+		log.debug("\tSorting lattice nodes in level {} by their dimensions " , latticeNodes.get(0).dimensions.size());
```

Comment:
To make the output more readable:). When traversing the lattice, detailed messages for each level are indented.